runtests.py

```diff
@@ -2372,6 +2372,7 @@ def runtests(options, cmd_args, coverage=None):
 
     if options.limited_api:
         CFLAGS.append("-DCYTHON_LIMITED_API=1")
```

Comment:
I think I tried to point this out on the old PR, but: at some point I think this should add `"-DPY_LIMITED_API"` too, since at present you're only checking that this is internally consistent and not that it actually conforms to the limited API. I don't think it needs to come with this change though - I just think it should be on the TODO list

Reply:
Totally! Slowly working towards that direction. Starting with `PY_LIMITED_API` just made the whole thing explode. I imagine that with these last couple of changes, it should now be manageable to start tackling the remaining parts to get the `PY_LIMITED_API` flag in!